models/resultModel.js

```diff
@@ -0,0 +1,33 @@
+import mongoose from 'mongoose';
+
+const resultSchema = new mongoose.Schema({
+
+    recordId: { type: Array, unique: true, required: true },
+    userId: { type: String, unique: true, required: true },
+    quizId: { type: String, unique: true, required: true },
+},
+    { timestamps: true },
+);
+
+
+const ResultModel = mongoose.model('Result', resultSchema);
+
+const save = async model => new ResultModel(model).save();
+
+
+const getAllResults = async () => ResultModel.find();
+
+const getResultById = async _id => ResultModel.findById({ _id });
+
+const getAllResultsByUserId = async userId => ResultModel.find({ userId });
+
+const getAllresultsByQuizId = async quizId => ResultModel.findOne({ quizId });
```

Comment:
There can be multiple results for one quize. findOne will return only one result